pkg/features/kube_features.go

```diff
@@ -433,6 +419,5 @@ var defaultKubernetesFeatureGates = map[utilfeature.Feature]utilfeature.FeatureS
 	apiextensionsfeatures.CustomResourceSubresources: {Default: true, PreRelease: utilfeature.Beta},
 
 	// features that enable backwards compatibility but are scheduled to be removed
-	ServiceProxyAllowExternalIPs: {Default: false, PreRelease: utilfeature.Deprecated},
-	ReadOnlyAPIDataVolumes:       {Default: true, PreRelease: utilfeature.Deprecated},
```

Comment:
I would expect a deprecated feature to default to false... what's the difference between a deprecated default-true feature gate and a GA feature?

Reply:
both of these were gates added around functional changes

the first likely should have been phrased to make the desired state going forward the positive (e.g. ServiceProxyRequirePodIP) and defaulted to true

The deprecated aspect was trying to indicate the *gate* was deprecated and the ability to change its default value would be removed shortly.